File: src/main/java/mnm/mods/tabbychat/util/MessagePatterns.java
package mnm.mods.tabbychat.util;

import static mnm.mods.tabbychat.util.Translation.*;
import net.minecraft.client.resources.I18n;

public enum MessagePatterns {

    ARROW(FORMAT_MESSAGE_ARROW,
            "^\\[" + MessagePatterns.PLAYER_PATTERN + "[ ]?\\-\\>[ ]?me\\]",
            "^\\[me[ ]?\\-\\>[ ]?" + MessagePatterns.PLAYER_PATTERN + "\\]"),
    TO_FROM(FORMAT_MESSAGE_TO_FROM,
            "^From " + MessagePatterns.PLAYER_PATTERN + "[ ]?:",
            "^To " + MessagePatterns.PLAYER_PATTERN + "[ ]?:"),
    WHISPERS(FORMAT_MESSAGE_WHISPER,
            "^" + MessagePatterns.PLAYER_PATTERN + " whispers to you:",
            "^You whisper to " + MessagePatterns.PLAYER_PATTERN + ":"),
    DISABLED(FORMAT_MESSAGE_DISABLED, "a^", "a^");

    private static final String PLAYER_PATTERN = "([\\w\\d_]{3,30})";

    private final String translation;
    private final String incoming;
    private final String outgoing;

    private MessagePatterns(String translation, String incoming, String outgoing) {
        this.translation = translation;
        this.incoming = incoming;
        this.outgoing = outgoing;
    }

    public String getIncoming() {
        return incoming;
    }

    public String getOutgoing() {
        return outgoing;
    }

    @Override
    public String toString() {
        return I18n.format(translation);
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/util/MessagePatterns.kt
package mnm.mods.tabbychat.util

enum class MessagePatterns(
        val display: Translatable,
        val incoming: String,
        val outgoing: String) {

    ARROW(Translation.FORMAT_MESSAGE_ARROW,
            "^\\[(.{3,30})[ ]?-\\>[ ]?me\\]",
            "^\\[me[ ]?-\\>[ ]?(.{3,30})\\]"),
    TO_FROM(Translation.FORMAT_MESSAGE_TO_FROM,
            "^From (.{3,30}) ?:",
            "^To (.{3,30}) ?:"),
    WHISPERS(Translation.FORMAT_MESSAGE_WHISPER,
            "^(.{3,30}) whispers to you:",
            "^You whisper to (.{3,30}):"),
    DISABLED(Translation.FORMAT_MESSAGE_DISABLED, "a^", "a^");
}
